eng/common/scripts/TypeSpec-Project-Generate.ps1

```diff
@@ -14,6 +14,78 @@ $ErrorActionPreference = "Stop"
 . $PSScriptRoot/common.ps1
 Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 
+function GetNpmPackageVersion([string]$packagePath) {
+    $packagePath = Resolve-Path $packagePath
+    $packageFolder = Split-Path -Parent $packagePath
+    $packageName = Split-Path -Leaf $packagePath
+    Push-Location $packageFolder
+    try {
+        $packageVersion = node -p -e "require('./$packageName').version"
+        return $packageVersion
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+function GetEmitterDependencyVersion([string]$packagePath) {
+    $emitterName = &$GetEmitterNameFn
+    $matchString = "`"$emitterName`": `"(.*?)`""
+    If ((Get-Content -Raw $packagePath) -match $matchString) {
+        return $Matches[1]
+    }
+}
+
+function NpmInstallAtRoot() {
+    $root = git rev-parse --show-toplevel
+    Push-Location $root
+    try {
+        #default to root/eng/emitter-package.json but you can override by writing
+        #Get-${Language}-EmitterPackageJsonPath in your Language-Settings.ps1
+        $replacementPackageJson = "$PSScriptRoot/../../emitter-package.json"
+        if (Test-Path "Function:$GetEmitterPackageJsonPathFn") {
+            $replacementPackageJson = &$GetEmitterPackageJsonPathFn
+        }
+
+        $emitterName = &$GetEmitterNameFn
+        $installedPath = Join-Path $root "node_modules" $emitterName "package.json"
+        if (Test-Path $installedPath) {
+            $installedVersion = GetNpmPackageVersion $installedPath
+            Write-Host "Installed emitter version: $installedVersion"
+
+            $replacementVersion = GetEmitterDependencyVersion $replacementPackageJson
```

Comment:
Why not pass in the emitterName here instead of having the function called again?